torchao/csrc/cuda/mx_kernels/mxfp8_quantize.cuh

```diff
@@ -900,10 +901,244 @@ __global__ void __launch_bounds__(MXFP8_THREADS_PER_CHUNK)
   // #endif
 }
 
+// 3D MXFP8 quantization kernel using 2D TMA
+template <typename IType, typename OType, size_t SCALE_DIM_Y,
+          size_t SCALE_DIM_X, ScaleCalculationMode ScalingMode>
+__global__ void __launch_bounds__(MXFP8_THREADS_PER_CHUNK)
+    mxfp8_quantize_kernel_3d(
+        const CUtensorMap* tensor_maps_input,
```

Comment:
Do we need to link against the cuda drivers in order to build this?

I think we fixed this at the top of the stack?

Reply:
do you mean in the create tensor map functions? the other examples I saw do reference the cuda driver to get the cuTensorMapEncodeTiled function pointer, so I did the same for 3d tensor map creation.